src/app_service/service/contacts/service.nim

```diff
@@ -767,34 +772,39 @@ QtObject:
 
   proc asyncContactInfoLoaded*(self: Service, pubkeyAndRpcResponse: string) {.slot.} =
     let rpcResponseObj = pubkeyAndRpcResponse.parseJson
-    let publicKey = $rpcResponseObj{"publicKey"}
-    let requestError = rpcResponseObj{"error"}
-    var error : string
-
-    if requestError.kind != JNull:
-      error = $requestError
-    else:
-      let responseError = rpcResponseObj{"response"}{"error"}
-      if responseError.kind != JNull:
-        error = Json.decode($responseError, RpcError).message
-
-    if len(error) != 0:
-      error "error requesting contact info", msg = error, publicKey
-      self.events.emit(SIGNAL_CONTACT_INFO_REQUEST_FINISHED, ContactInfoRequestArgs(publicKey: publicKey, ok: false))
-      return
+    let publicKey = rpcResponseObj{"publicKey"}.getStr
+    try:
+      let requestError = rpcResponseObj{"error"}
+      var error : string
 
-    let contact = rpcResponseObj{"response"}{"result"}.toContactsDto()
-    self.saveContact(contact)
-    self.events.emit(SIGNAL_CONTACT_INFO_REQUEST_FINISHED, ContactInfoRequestArgs(publicKey: publicKey, ok: true))
+      if requestError.kind != JNull:
+        error = $requestError
+      else:
+        let responseError = rpcResponseObj{"response"}{"error"}
+        if responseError.kind != JNull:
+          error = Json.decode($responseError, RpcError).message
 
-  proc requestContactInfo*(self: Service, pubkey: string) =
-    try:
-      let arg = AsyncRequestContactInfoTaskArg(
-        tptr: cast[ByteAddress](asyncRequestContactInfoTask),
-        vptr: cast[ByteAddress](self.vptr),
-        slot: "asyncContactInfoLoaded",
-        pubkey: pubkey,
-      )
-      self.threadpool.start(arg)
+      if len(error) != 0:
+        raise newException(CatchableError, error)
+
+      var contact = rpcResponseObj{"response"}{"result"}.toContactsDto()
+
+      let foundTheContactInfo = contact.displayName != ""
+      self.updateContact(contact)
+      self.events.emit(SIGNAL_CONTACT_INFO_REQUEST_FINISHED,
+        ContactInfoRequestArgs(publicKey: publicKey, ok: foundTheContactInfo))
     except Exception as e:
-      error "Error requesting contact info", msg = e.msg, pubkey
+      error "error requesting contact info", msg = e.msg, publicKey
+      self.events.emit(SIGNAL_CONTACT_INFO_REQUEST_FINISHED, ContactInfoRequestArgs(publicKey: publicKey, ok: false))
+      var contact = self.getContactById(publicKey)
+      contact.fetching = false
+      self.updateContact(contact)
+
+  proc requestContactInfo*(self: Service, pubkey: string) =
```

Comment:
Do you think we could (or even should) set `fetching = true` here, to make the state of `fetching` correct even with direct calls of `requestContactInfo`?